mteb/evaluate.py

```diff
@@ -265,12 +271,6 @@ def evaluate(
     if encode_kwargs is None:
         encode_kwargs = {}
```

Comment:
There is a chance that encode_kwargs is None at line 109 when we check for `if "batch_size" not in encode_kwargs:`. So maybe this highlighted check should be moved up as well?

Reply:
It happens in two different functions now:
`evaluate` > none check (to ensure that `_evaluate` always gets a dict)
`_evaluate` > add batch_size, but only if needed

(renamed `_evaluate` to `_evaluate_task` to make the two more distinct)

Moved it all into `evaluate` now, as I discovered that the current implementation would give you the batch size logging for every task. It should not only log once.